pkg/yurthub/certificate/pki.go

```diff
@@ -41,22 +103,9 @@ func GenTLSConfigUseCertMgrAndCertPool(
 		ClientAuth: tls.VerifyClientCertIfGiven,
 	}
 
-	tlsConfig.GetClientCertificate =
-		func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			cert := m.Current()
-			if cert == nil {
-				return &tls.Certificate{Certificate: nil}, nil
-			}
-			return cert, nil
-		}
-	tlsConfig.GetCertificate =
-		func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
-			cert := m.Current()
-			if cert == nil {
-				return &tls.Certificate{Certificate: nil}, nil
-			}
-			return cert, nil
-		}
+	scm := newHubServerCertMgr(certDir, m)
+	tlsConfig.GetClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) { return scm.Current(), nil }
```

Comment:
maybe `GetClientCertificate` will not used by `yurthub https server`

Reply:
I think it's right, according to the local test result, this function will not be used by yurthub https server, so I delete the tlsConfig.GetClientCertificate, please take review.